src/main/kotlin/org/jitsi/jicofo/JicofoServices.kt

```diff
@@ -94,20 +94,24 @@ open class JicofoServices(
             }
         }
 
-        val focusJid = XmppConfig.client.username.toString() + "@" + XmppConfig.client.domain.toString()
         conferenceRequestHandler = ConferenceRequestHandler(
             focusManager = focusManager,
-            focusAuthJid = focusJid,
+            focusAuthJid = XmppConfig.client.username.toString() + "@" + XmppConfig.client.domain.toString(),
             isFocusAnonymous = StringUtils.isBlank(XmppConfig.client.password),
             authAuthority = authenticationAuthority,
             reservationSystem = reservationSystem
         )
-        focusComponent = FocusComponent(XmppComponentConfig.config, conferenceRequestHandler).apply {
-            val configService = ServiceUtils2.getService(bundleContext, ConfigurationService::class.java)
-            loadConfig(configService, "org.jitsi.jicofo")
-            authenticationAuthority?.let { setAuthAuthority(authenticationAuthority) }
+
+        focusComponent = if (XmppComponentConfig.config.enabled) {
+            FocusComponent(XmppComponentConfig.config, conferenceRequestHandler).apply {
+                val configService = ServiceUtils2.getService(bundleContext, ConfigurationService::class.java)
+                loadConfig(configService, "org.jitsi.jicofo")
+                authenticationAuthority?.let { setAuthAuthority(authenticationAuthority) }
+                connect()
```

Comment:
Reading this I wonder if it would be better to move the `connect` call out to make it more obvious (also since the code in the apply is more about initialization and this is a bit different, so doing:

```
focusComponent = if (...) {
    ...
} else {
    null
}

focusComponent?.connect()
```